src/flags.rs

```diff
@@ -0,0 +1,65 @@
+#[derive(Default)]
+pub struct X86Flags {
+	pub carry: bool,
+	pub parity: bool,
+	pub adjust: bool,
+	pub zero: bool,
+	pub sign: bool,
+	pub trap: bool,
+	pub interrupt: bool,
+	pub direction: bool,
+	pub overflow: bool,
+	pub iopl: bool,
+	pub nested_task: bool,
+	pub resume: bool,
+	pub virtual_8086_mode: bool,
+	pub alignment_check: bool,
+	pub id_capability: bool,
+	pub vif: bool,
+	pub vip: bool
+}
+
+impl X86Flags {
+	pub fn calculate_parity(&mut self, val: u32) {
+		let mut count = 0;
+		for i in 0..7 {
+			if (val&((1<<i)))!=0{
+				count+=1;
+			}
+		}
+		self.parity = (count % 2) == 0;
+	}
+	pub fn calculate_sign8(&mut self, val: u8) {
+		if (val & 0x80) == 0 {
```

Comment:
This can be simplified to `self.sign = ((val & 0x80) == 0)`, though that's somewhat just a style thing